hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/KeyManagerImpl.java

```diff
@@ -405,6 +405,9 @@ private OmKeyInfo readKeyInfo(OmKeyArgs args) throws IOException {
         value = getOmKeyInfoFSO(volumeName, bucketName, keyName);
       } else {
         value = getOmKeyInfoDirectoryAware(volumeName, bucketName, keyName);
+        if (bucketLayout.isLegacy() && value != null && !value.isFile()) {
```

Comment:
Is it right to put the check here? I think it's an S3 logic. OM always returns key requests, and it's up to the clients (S3 or OFS) to determine how the result is intercepted.

Reply:
Yes, that is true, but the intention here is not to change the behavior of LEGACY buckets. I considered doing it only in S3G, but there are 2 issues there, it (keeps the regression) changes the behavior of LEGACY without documenting and communicating it widely. Secondly, detecting directory objects is much more involved at the S3G level.
There need to be more work as part of the OBS and  FSO bucket evolution to address these issues. For now, let's keep LEGACY the way it was.